src/dispatch/notification/models.py

```diff
@@ -0,0 +1,82 @@
+from datetime import datetime
+from typing import Optional, List
+
+from sqlalchemy import Column, Integer, String, ForeignKey, Table, PrimaryKeyConstraint
+from sqlalchemy.orm import relationship
+from sqlalchemy_utils import TSVectorType
+
+from dispatch.database import Base
+from dispatch.incident_priority.models import (
+    IncidentPriorityCreate,
+    IncidentPriorityRead,
+    IncidentPriorityUpdate,
+)
+from dispatch.incident_type.models import IncidentTypeCreate, IncidentTypeRead, IncidentTypeUpdate
+
+from dispatch.models import DispatchBase, TimeStampMixin
+
+
+# Association tables for many to many relationships
+assoc_notification_incident_types = Table(
+    "notification_incident_type",
+    Base.metadata,
+    Column("incident_type_id", Integer, ForeignKey("incident_type.id", ondelete="CASCADE")),
+    Column("notification_id", Integer, ForeignKey("notification.id", ondelete="CASCADE")),
+    PrimaryKeyConstraint("incident_type_id", "notification_id"),
+)
+
+assoc_notification_incident_priorities = Table(
+    "notification_incident_priority",
+    Base.metadata,
+    Column("incident_priority_id", Integer, ForeignKey("incident_priority.id", ondelete="CASCADE")),
+    Column("notification_id", Integer, ForeignKey("notification.id", ondelete="CASCADE")),
+    PrimaryKeyConstraint("incident_priority_id", "notification_id"),
+)
+
+
+class Notification(Base, TimeStampMixin):
+    # Columns
+    id = Column(Integer, primary_key=True)
+    conversation_name = Column(String)
```

Comment:
Should this be more general? I think we could use `service` as a good template: https://github.com/Netflix/dispatch/blob/master/src/dispatch/service/models.py#L53

That way we could use these to define email notifications as well as slack conversations.

Reply:
Yes, I was thinking we could allow users to create notifications and allow them to select from a list of already defined notification (or message types) and plugins that provide communications-related functionality (e.g. email, conversation), and then, depending on the plugin selected, you would provide where that notification has to go. For conversation notifications, we could support both channels and users.

Reply:
The other portion is to allow a user to define a subset of incidents to be notified for. The service model allows terms, but for notifications, I assume we would also want to allow tags to be associated with notifications such that, when an incident has a given tag that notification will be sent. 

For example: userFooNotification(tag1 OR tag2) -- incidentBar(tag2) -- email -> send email notification